extensions/stackdriver/log/exporter.cc

```diff
@@ -70,9 +71,22 @@ ExporterImpl::ExporterImpl(RootContext* root_context,
   if (logging_service_endpoint.empty()) {
     grpc_service.mutable_google_grpc()->set_target_uri(
         kGoogleStackdriverLoggingAddress);
-    grpc_service.mutable_google_grpc()
-        ->add_call_credentials()
-        ->mutable_google_compute_engine();
+    if (sts_port.empty()) {
+      // Security token exchange is not enabled. Use default GCE credential.
+      grpc_service.mutable_google_grpc()
+          ->add_call_credentials()
+          ->mutable_google_compute_engine();
+    } else {
+      auto* sts_service = grpc_service.mutable_google_grpc()
+                              ->add_call_credentials()
+                              ->mutable_sts_service();
+      sts_service->set_token_exchange_service_uri(
+          "http://127.0.0.1:" + sts_port + "/token");
+      sts_service->set_subject_token_path("/tmp/sts-envoy-token.jwt");
```

Comment:
The token path is /var/run/secrets/tokens/istio-token https://github.com/istio/istio/blob/b28b5a0b2749c2ad8a50c82826860a3183653d9c/pilot/pkg/security/model/authentication.go#L46

Reply:
Done.